### ISSUE TITLE
[BUG] Volume Gesture

### ISSUE THREAD

**Issue by anuraag488** [anilbeesetti/nextplayer#637]
**Describe the bug**
Volume Gesture not working on right portion of screen. I have to swipe somewhere near to centre of screen to make it work while brightness gesture works fine on left side. In attached screenshot highlighted area gesture doesn't work.

**Expected behavior**

**Screenshots**

![Screenshot_20230926-094140_Next Player~2](https://github.com/anilbeesetti/nextplayer/assets/7353086/94e5918c-6bbc-4901-b7bb-6fe56191af7c)

**Device info (please complete the following information):**
 - Device: Redmi Note 9 Pro Max
 - Android version: Lineageos 20 Android 13
 - App version: 0.9.0

**Additional context**

**Comment by anilbeesetti** [comment]
@anuraag488. Is this issue present in previous versions? If possible send screen recording

**Comment by anuraag488** [comment]
I wasn't using next player. I have tried a 0.7.0 just now. Same issue. Happens with landscape orientation.

**Comment by Sureshgorai25** [comment]
Fine for me, no issues

**Comment by Soulavanger** [comment]
Volume gesture is working fine but after changing tabs or doing something else and then coming back to the player after that right side gestures are not working especially volume gesture

**Comment by Soulavanger** [comment]
https://github.com/anilbeesetti/nextplayer/assets/121586876/548fc3b9-55b1-441c-a7f7-db4a29a30794

This is the video regarding the volume issue

**Comment by drogga** [comment]
_Since I have some feedback regarding the volume - I'm posting it here, because IDT it makes sense to open another one that's related. If it's a problem and not the correct place - I will delete this comment, don't worry._

When I pause & play back, I can hear that the volume (set to 100%) is being gain-ed / boosted / amplified for a second, haven't tested with headset, but phone's speakers only.

The 100% volume in the app is out of sync by 1 step with the system (media ofc.) one, when I set it to 100 percent in the app, the system one has one more step to 100, the app sets it to about ~97+/-, the system 100% in the app means like ~102/3.

IDK if the app remembers the boost level, but it would probably be nice to have a "Remember Volume" option in the settings\preferences, like the brightness one for example.

I wish the app supported Picture-in-picture (PiP > it's Planned, I know) and especially background playback ("appear on top" permission is probably needed), so the video would continue playing when it's not in focus or even when the device/screen is locked (not the app's controls locked), this bug in the video above when returning to the app, probably wouldn't happen if the playback didn't stop.

I prefer if the volume indicator is on the right side (even if the finger covers it when swiping).

I would like if the options to disable the brightness and volume gesture are separate ones, so I can only disable the brightness one, so when I swipe-up on the left side nothing to happen.

Lineage 18.1 (=A. 11).

**Comment by anuraag488** [comment]
> VID_20230928205510.mp4
> 
> This is the video regarding the volume issue

@anilbeesetti this is exactly how volume gesture doesn't work. In my case I don't need to switch to other app.

**Comment by anilbeesetti** [comment]
> _Since I have some feedback regarding the volume - I'm posting it here, because IDT it makes sense to open another one that's related. If it's a problem and not the correct place - I will delete this comment, don't worry._
> 
> When I pause & play back, I can hear that the volume (set to 100%) is being gain-ed / boosted / amplified for a second, haven't tested with headset, but phone's speakers only.
> 
> The 100% volume in the app is out of sync by 1 step with the system (media ofc.) one, when I set it to 100 percent in the app, the system one has one more step to 100, the app sets it to about ~97+/-, the system 100% in the app means like ~102/3.
> 
> IDK if the app remembers the boost level, but it would probably be nice to have a "Remember Volume" option in the settings\preferences, like the brightness one for example.
> 
> I wish the app supported Picture-in-picture (PiP > it's Planned, I know) and especially background playback ("appear on top" permission is probably needed), so the video would continue playing when it's not in focus or even when the device/screen is locked (not the app's controls locked), this bug in the video above when returning to the app, probably wouldn't happen if the playback didn't stop.
> 
> I prefer if the volume indicator is on the right side (even if the finger covers it when swiping).
> 
> I would like if the options to disable the brightness and volume gesture are separate ones, so I can only disable the brightness one, so when I swipe-up on the left side nothing to happen.
> 
> Lineage 18.1 (=A. 11).

Sorry for late reply @drogga. I will try to address these issues soon

**Comment by drogga** [comment]
If you used VLC (with the vol. loudness enhancer / boost/amp/gain enabled), you've probably noticed that there's a safety stopper to/at 100%, showing the following toast msg.: "Slide up again to go beyond 100%", so you can't accidentally trigger the boost by fast swiping up once, thinking that you will set it to 100%, but actually boosting it beyond that - maybe this is something to consider implementing in future in this awesome player. ;)

Edit:
If you end up making it to work the same as VLC, then it doesn't have to annoy/nag with a toast msg. that much everytime, like VLC does, but just only once - the first time the user swipes up fast, just to get an idea what's going on and how to trigger the boost from now on and to not show up / notify ever again.

**Comment by Yashraj254** [comment]
@anilbeesetti i was facing the same issue in one of my device, i looked into it and figured it out that its happening because of the inExclusionArea() method of PlayGestureHelper 

its because of different insets values in different devices (i am not very familiar with these functionalities)

these are the insets in two of my devices:

1.  Oneplus 6t(Width: 2260 , Height: 1080) Landscape 
Insets - Left: 0, Right: 0, Top: 80, Bottom: 0            (No issue with getures)

2. Oneplus Nord(Width: 2285, Height: 1032) Landscape
Insets - Left: 205, Right: 90, Top: 115, Bottom: 96  (Minor issue with volume and brightness gesture) 

so this condition returns true in slightly different areas in different devices.
(firstEvent.x < insets.left) || (firstEvent.x > (screenWidth - insets.right)) || (firstEvent.y < insets.top) || (firstEvent.y > (screenHeight - insets.bottom))

maybe we can remove this method or it might cause any issues(not sure about that, i didnt face any )

**Comment by anilbeesetti** [comment]
@Yashraj254 What's the android version of both devices? And are you using full screen gestures or soft buttons for navigation?

**Comment by Yashraj254** [comment]
@anilbeesetti 
Oneplus 6t - Android 11
Oneplus Nord - Android 13
Full Screen gestures for navigation in both devices

**Comment by anilbeesetti** [comment]
> @anilbeesetti i was facing the same issue in one of my device, i looked into it and figured it out that its happening because of the inExclusionArea() method of PlayGestureHelper
> 
> its because of different insets values in different devices (i am not very familiar with these functionalities)
> 
> these are the insets in two of my devices:
> 
>     1. Oneplus 6t(Width: 2260 , Height: 1080) Landscape
>        Insets - Left: 0, Right: 0, Top: 80, Bottom: 0            (No issue with getures)
> 
>     2. Oneplus Nord(Width: 2285, Height: 1032) Landscape
>        Insets - Left: 205, Right: 90, Top: 115, Bottom: 96  (Minor issue with volume and brightness gesture)
> 
> 
> so this condition returns true in slightly different areas in different devices. (firstEvent.x < insets.left) || (firstEvent.x > (screenWidth - insets.right)) || (firstEvent.y < insets.top) || (firstEvent.y > (screenHeight - insets.bottom))
> 
> maybe we can remove this method or it might cause any issues(not sure about that, i didnt face any )

If we remove this method then player gestures like volume, brightness, seek gestures conflict with system gestures. Like if you want to go back by swiping on the left or right edge of the screen, it would trigger seek gesture and in the same way if you swipe from bottom to go to home screen, it would trigger volume or brightness gesture

**Comment by Yashraj254** [comment]
@anilbeesetti yeah you are right, however if we dont subtract the insets from screen width, it reduces the excluded area, and not really causing conflicting issues(try it once)

**Comment by anilbeesetti** [comment]
@anuraag488 @Sureshgorai25 @Yashraj254 Could you test the fix. Here's the apk:- https://github.com/anilbeesetti/nextplayer/suites/17060592559/artifacts/974491088

**Comment by anuraag488** [comment]
> @anuraag488 @Sureshgorai25 @Yashraj254 Could you test the fix. Here's the apk:- https://github.com/anilbeesetti/nextplayer/suites/17060592559/artifacts/974491088

Same for me. Right side now working.

https://github.com/anilbeesetti/nextplayer/assets/7353086/a439cf2e-b31c-47a1-95bb-5523e946d1e8

**Comment by Sureshgorai25** [comment]
> @anuraag488 @Sureshgorai25 @Yashraj254 Could you test the fix. Here's the apk:- https://github.com/anilbeesetti/nextplayer/suites/17060592559/artifacts/974491088

I don't have any issues with the volume gesture, it's fine for me

**Comment by anuraag488** [comment]
Tested on 2 more device (Samsung S21 Stock Rom and Redmi Note 9 Pro Max Stock MIUI). Same issue.


https://github.com/anilbeesetti/nextplayer/assets/7353086/147f8d44-b2fe-4c85-94d3-60dee79afce8

**Comment by ngocanhtve** [comment]
My Sony running Android 13 custom rom, hasn't this bugs.

**Comment by drogga** [comment]
As I understand it - that place is usually taken by the NavBar, that's why it doesn't work at the very edge of the screen, even if you use gestures instead of 3 buttons for the system navigation and if you rotate the device on the opposite side (landscape > sensor), then the brightness swipe-up gesture probably won't work from the very edge of the screen.

**Comment by anuraag488** [comment]
> As I understand it - that place is usually taken by the NavBar, that's why it doesn't work at the very edge of the screen, even if you use gestures instead of 3 buttons for the system navigation and if you rotate the device on the opposite side (landscape > sensor), then the brightness swipe-up gesture probably won't work from the very edge of the screen.

Gesture for players like MxPlayer and Xplayer works on both system bars. For JustPlayer Gesture doesn't work on navigation bar area but works on status bar.

But on NextPlayer volume gesture is too far from navigation bar height (2-3x times) from where gesture work for me. Brightness gesture works fine on both landscape orientation. Look into below video

https://github.com/anilbeesetti/nextplayer/assets/7353086/6833676b-1deb-4e8b-9a1a-6b8c4e7d9dda

**Comment by Yashraj254** [comment]
> @anuraag488 @Sureshgorai25 @Yashraj254 Could you test the fix. Here's the apk:- https://github.com/anilbeesetti/nextplayer/suites/17060592559/artifacts/974491088

@anilbeesetti issue still remains..
